x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -228,34 +234,59 @@ export const useIndexFields = (sourcererScopeName: SourcererScopeName) => {
     [dispatch, sourcererScopeName]
   );
 
+  const getSignalIndexNameSelector = useMemo(
+    () => sourcererSelectors.signalIndexNameSelector(),
+    []
+  );
+  const signalIndexNameSelector = useDeepEqualSelector(getSignalIndexNameSelector);
+
   const indexFieldsSearch = useCallback(
-    (indicesName) => {
+    (selectedDataViewId: string, newSignalsIndex?: string) => {
       const asyncSearch = async () => {
         abortCtrl.current = new AbortController();
         setLoading(true);
+
         searchSubscription$.current = data.search
-          .search<IndexFieldsStrategyRequest, IndexFieldsStrategyResponse>(
-            { indices: indicesName, onlyCheckIfIndicesExist: false },
+          .search<IndexFieldsStrategyRequest<'dataView'>, IndexFieldsStrategyResponse>(
+            {
+              dataViewId: selectedDataViewId,
+              onlyCheckIfIndicesExist: false,
+            },
             {
               abortSignal: abortCtrl.current.signal,
               strategy: 'indexFields',
             }
           )
           .subscribe({
             next: (response) => {
+              // TODO: Steph/sourcerer needs better tests
               if (isCompleteResponse(response)) {
-                const stringifyIndices = response.indicesExist.sort().join();
+                const newSelectedPatterns = selectedPatterns.filter((pattern) =>
+                  patternList.includes(pattern)
+                );
+                const patternString = newSelectedPatterns.sort().join();
+                const signalIndexName = signalIndexNameSelector ?? newSignalsIndex;
                 dispatch(
                   sourcererActions.setSource({
                     id: sourcererScopeName,
                     payload: {
-                      browserFields: getBrowserFields(stringifyIndices, response.indexFields),
-                      docValueFields: getDocValueFields(stringifyIndices, response.indexFields),
+                      // TODO: Steph/sourcerer all the below formatting should be happening serverside
```

Comment:
I'm also going to move these fields from the `sourcerer.sourcererScopes[scopeName]` to the `sourcerer.kibanaDataViews` so we don't duplicate requests for timeline/default/detections when they are using the same data view https://github.com/elastic/security-team/issues/1730